scanner/mappers/mappers.go

```diff
@@ -788,3 +788,65 @@ func findName(vuln *claircore.Vulnerability, p *regexp.Regexp) (string, bool) {
 	}
 	return "", false
 }
+
+// filterVulns filters repeat vulnerabilities out of vulnIDs and returns the result.
+// This function does not guarantee ordering is preserved.
+func filterVulns(vulnIDs []string, vulns map[string]*claircore.Vulnerability) []string {
+	// Group each vulnerability by name.
+	// This maps each name to a slice of vulnerabilities to protect against the possibility
+	// Claircore finds multiple vulnerabilities with the same name for this package from different vulnerability streams.
+	// In that situation, it is not clear which one may be the single, correct option to choose, so just allow for both.
+	vulnsByName := make(map[string][]*claircore.Vulnerability)
+OUTER:
+	for _, vulnID := range vulnIDs {
+		vuln := vulns[vulnID]
+		if vuln == nil {
+			continue
+		}
+
+		// Find the currently tracked vulnerabilities with the same name.
+		// If this entry matches any of those, then ignore this one.
+		matching := vulnsByName[vuln.Name]
+		for _, match := range matching {
+			if vulnsEqual(match, vuln) {
+				continue OUTER
+			}
+		}
+
+		// Add the unique entry to the map.
+		vulnsByName[vuln.Name] = append(vulnsByName[vuln.Name], vuln)
+	}
+
+	filtered := make([]string, 0, len(vulns))
+	for _, vulns := range vulnsByName {
+		for _, vuln := range vulns {
+			filtered = append(filtered, vuln.ID)
+		}
+	}
+	return filtered
+}
+
+// vulnsEqual determines if the vulnerabilities are essentially equal.
+// That is, this function does not check all fields of the vulnerability struct,
+// to prevent consumers from seeing two seemingly identical vulnerabilities
+// for the same package in the same image.
+//
+// For example: Claircore currently returns CVE-2019-12900 twice for the bzip2-libs package
+// in one particular image. The two versions of the CVE are exactly the same
+// except for the repository name (cpe:/a:redhat:enterprise_linux:8::appstream vs cpe:/o:redhat:enterprise_linux:8::baseos).
+// The entry for this vulnerability as it matched this package in this image may be found in
+// https://access.redhat.com/security/data/oval/v2/RHEL8/rhel-8-including-unpatched.oval.xml.bz2.
+// After reading the entry in this file, it is clear Claircore matched this vulnerability to this stream's
+// CVE-2019-12900 entry twice (once per matching repository).
+//
+// The goal of this function is to make it clear those two CVE-2019-12900 findings are exactly the same.
+func vulnsEqual(a, b *claircore.Vulnerability) bool {
+	return a.Updater == b.Updater &&
```

Comment:
`Updater` doesn't appear exposed to clients of Scanner V4, couldn't this lead to scenario where there are two identical vulns are in a vuln report?

Reply:
it doesn't, as you mentioned, but this is protecting us against something like finding a go vuln in both a Red hat stream and OSV (possible? maybe?). In that case, I think it's ok to show both.

This PR really just protects us against finding the same vuln twice in the same updater

Reply:
You know what, if the Updater is hidden, and the rest are the same, then no reason to see them as the same. I'll change